engine-plugins/frame-plugins/src/main/scala/org/trustedanalytics/atk/engine/frame/plugins/load/JdbcPlugin/JdbcConnectionFactory.scala

```diff
@@ -0,0 +1,99 @@
+/*
+// Copyright (c) 2015 Intel Corporation 
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+package org.trustedanalytics.atk.engine.frame.plugins.load.JdbcPlugin
+
+import java.sql.{ DriverManager, Connection }
+import org.trustedanalytics.atk.domain.frame.load.JdbcArgs
+
+/**
+ * Jdbc factory for connections and connections strings
+ */
+object JdbcConnectionFactory extends Serializable {
+
+  /**
+   * Create a new connection to database
+   * @param arguments connection required arguments
+   * @param user user
+   * @param password user password
+   * @return a jdbc connection
+   */
+  def createConnection(arguments: JdbcArgs, user: String, password: String): Connection = {
+
+    //load the driver
+    Class.forName(arguments.driverName)
+    DriverManager.getConnection(connectionUrl(arguments), user, password)
+  }
+
+  /**
+   * Create a connection url string for jdbc
+   * @param arguments required arguments
+   * @return a connection url
+   */
+  def connectionUrl(arguments: JdbcArgs): String = {
+
+    var url: String = "jdbc:"
+
+    //connectionUrl
+    if (isMySql(arguments.driverType)) {
```

Comment:
You could probably have a DriverType algebraic datatype (sealed trait DriverType with two case objects that extend it, one for mysql and one for sql server). That way the trait could define a connectionUrl method, and each drivertype could implement the method as appropriate. That way you could avoid all these if statements and just say: val url = arguments.driverType.connectionUrl(arguments). Plus you'd get the validation for only using supported driver types for free.